src/api/define.js

```diff
@@ -97,19 +97,28 @@ function createInitProps (Ctor) {
 }
 
 function generateUniqueName(name) {
-  const registered = registry[name];
-  return registered ? `${name}-${registered}` : name;
-}
+  // we don't need to generate a unique name if it's the first time
+  if (!registry[name]) {
+    registry[name] = true;
+    return name;
+  }
+  // copy-pasted from here, looks like a good solution : http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript/2117523#2117523
```

Comment:
Uber nit: it'd be nice to clean up the function a little bit. `const`, `===`, `() => {}`; the works ;). Maybe make the comment just the URL.